lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js

```diff
@@ -123,13 +140,15 @@ class OptimizedImages extends Gatherer {
     const networkRecords = traceData.networkRecords;
     const imageRecords = OptimizedImages.filterImageRequests(options.url, networkRecords);
 
-    return Promise.all(imageRecords.map(record => {
-      return this.calculateImageStats(options.driver, record).catch(err => {
-        return {failed: true, err};
-      }).then(stats => {
-        return Object.assign(stats, record);
+    return imageRecords.reduce((promise, record) => {
+      return promise.then(results => {
+        return this.calculateImageStats(options.driver, record).catch(err => {
+          return {failed: true, err};
+        }).then(stats => {
+          return results.concat(Object.assign(stats, record));
+        });
       });
-    })).then(results => {
+    }, Promise.resolve([])).then(results => {
```

Comment:
nit: it might be nice to break this reduce up from the `.then` chain. There's already a lot going on. You could also remove some of the `{ return ...}` in the arrow functions and just do an implicit return.